src/MappedArrays.jl

```diff
@@ -62,9 +62,10 @@ parenttype{T,N,A,F,Finv}(::Type{MappedArray{T,N,A,F,Finv}}) = A
 
 function testvalue(data)
     if !isempty(data)
-        return first(data)
-    end
-    zero(eltype(data))
+        first(data)
+    else
+        zero(eltype(data))
+    end::eltype(data)
```

Comment:
That is a good point, I will think on that a little.

Aside from that it doesn't seem like this annotation made it worse than it was, since `first` ruined the party already

```julia
julia> data = Real[3.2]
julia> @code_warntype first(data)
Variables:
  #self#::Base.#first
  a::Array{Real,1}

Body:
  begin 
      # meta: location abstractarray.jl eachindex 679
      # meta: location abstractarray.jl indices1 74
      # meta: location abstractarray.jl indices 67
      SSAValue(2) = (Base.arraysize)(a::Array{Real,1},1)::Int64
      # meta: location tuple.jl map 92
      SSAValue(1) = SSAValue(2)
      # meta: pop location
      # meta: pop location
      # meta: pop location
      # meta: pop location
      return (Base.arrayref)(a::Array{Real,1},1)::Real
  end::Real
```